source/uwp/Renderer/lib/AdaptiveImageRenderer.cpp

```diff
@@ -321,32 +328,55 @@ namespace AdaptiveCards::Rendering::Uwp
+                                            winrt::IAsyncOperation<winrt::SvgImageSourceLoadStatus> const& operation, winrt::AsyncStatus status)
                                         {
-                                            if (auto strongThis = weakThis.get())
+                                            auto loadStatus = operation.GetResults();
+                                            if (status == winrt::AsyncStatus::Completed && isAutoSize &&
+                                                loadStatus == winrt::SvgImageSourceLoadStatus::Success)
                                             {
-                                                strongThis->SetAutoSize(uiElement, parentElement, imageContainer, isVisible, false /* imageFiresOpenEvent */);
+                                                if (auto strongThis = weakThis.get())
+                                                {
+                                                    strongThis->SetAutoSize(uiElement, parentElement, imageContainer, isVisible, false /* imageFiresOpenEvent */);
+                                                }
                                             }
-                                        }
-                                    });
+                                        });
+                                }
+                                else
+                                {
+                                    auto setSourceAction = image.as<winrt::BitmapImage>().SetSourceAsync(randomAccessStream);
+
+                                    setSourceAction.Completed(
+                                        [weakThis, uiElement, isAutoSize, parentElement, imageContainer, isVisible](
+                                            winrt::IAsyncAction const&, winrt::AsyncStatus status)
+                                        {
+                                            if (status == winrt::AsyncStatus::Completed && isAutoSize)
+                                            {
+                                                if (auto strongThis = weakThis.get())
+                                                {
+                                                    strongThis->SetAutoSize(uiElement, parentElement, imageContainer, isVisible, false /* imageFiresOpenEvent */);
+                                                }
+                                            }
+                                        });
+                                }
                             }
                         }
                         else
                         {
                             if (auto strongThis = weakThis.get())
                             {
-                                this->m_imageLoadTracker->MarkFailedLoadBitmapImage(bitmapImage);
+                                // Question: should we only mark as a failed image if (!m_enableXamlImageHandling && (m_listeners.size() != 0))
```

Comment:
Should we only mark as a failed image if (!m_enableXamlImageHandling && (m_listeners.size() != 0))?

Reply:
m_enableXamlImageHangling should be precursor to completion block being called?
m_listeners.size() should be checked by image load tracker?

Reply:
@beervoley Do you have any thoughts on this?